homeassistant/components/prowl/notify.py

```diff
@@ -80,3 +85,68 @@ async def async_send_message(self, message: str, **kwargs: Any) -> None:
                 ) from ex
             _LOGGER.error("Unexpected error when calling Prowl API: %s", str(ex))
             raise HomeAssistantError("Unexpected error when calling Prowl API") from ex
+
+
+class ProwlNotificationEntity(NotifyEntity):
+    """Implement the notification service for Prowl.
+
+    This class is used for Prowl config entries.
+    """
+
+    def __init__(self, hass: HomeAssistant, name: str, api_key: str) -> None:
+        """Initialize the service."""
+        self._hass = hass
+        self._prowl = prowlpy.Prowl(api_key)
+        self._attr_name = name
+        self._attr_unique_id = name
+
+    async def async_verify_key(self) -> bool:
+        """Validate API key."""
+        try:
+            async with asyncio.timeout(10):
+                await self._hass.async_add_executor_job(self._prowl.verify_key)
+                return True
+        except prowlpy.APIError as ex:
+            if str(ex).startswith("Invalid API key"):
+                return False
+            raise
```

Comment:
This should not be a method on the entity class. Instead, move it to `prowl/helpers.py` and call it from `async_setup_entry` and from the config flow.